Constructing shared CARingBuffer cannot fail

kkinnunen-apple opened the pull request:
#### eece78fdf9b27b52c5bbfca25a207495098ce8c7
<pre>
Constructing shared CARingBuffer cannot fail
<a href="https://bugs.webkit.org/show_bug.cgi?id=247360">https://bugs.webkit.org/show_bug.cgi?id=247360</a>
rdar://problem/101856691

Reviewed by Eric Carlson.

This is a problem because shared CARingBuffers are used cross-process,
and the WP might send data that would construct a potentially invalid
CARingBuffer. This leads to code that uses empty instances instead of
not checking all use-sites correctly. This in turn leads to running
code that is expected to only run when the ring buffer is correct.

Instead, make ConsumerCARingBuffer construction failable and hold
the instance accordingly. This is similar to many other related,
regular WebKit concepts like SharedMemory. Check all the construction
sites and use sites to not start audio processing unless the ring buffer is
present. This makes it easier in future to add more IPC validation
to the call sites that pass the ring buffer around and ensure that
the other parts of the code are consistent.

Previously, CARingBuffer was non-polymorphic class and its storage
class CARingBufferStorage was the polymorphic base class.
The usage was:
 CARingBuffer with CARingBufferStorageVector for in process ring buffer
 CARingBuffer with ReadOnlyRingBufferStorage for GPUP side consumer
 CARingBuffer with SharedRingBufferStorage for WP side producer

The two-class split does not make sense, as there are only one
non-polymorphic class and the arity is 1:1 in all cases.

Instead, rename:
 CARingBuffer is polymorphic base class
 InProcessCARingBuffer is the in-process ring buffer
 ProducerSharedCARingBuffer is the WP producer
 ConsumerSharedCARingBuffer is the read only GPUP side consumer

Moves the WebKit only code, CARingBuffer::adoptStorage() to
ConsumerSharedCARingBuffer::map().

Make sure Consumer is not superclass of Producer, as they&apos;re not
substitutable.

Use type alias ConsumerSharedCARingBuffer::Handle to communicate
the ring buffer to the receiver. This is so that it is obvious
what is being transferred as well as future-proofing the case
in which transferring ConsumerSharedCARingBuffer would need
more than the SHM handle. Use rvalue references to pass
the handle.

* Source/WebCore/platform/audio/cocoa/AudioSampleDataSource.h:
* Source/WebCore/platform/audio/cocoa/AudioSampleDataSource.mm:
(WebCore::AudioSampleDataSource::AudioSampleDataSource):
(WebCore::AudioSampleDataSource::setOutputFormat):
(WebCore::AudioSampleDataSource::pushSamplesInternal):
(WebCore::AudioSampleDataSource::pullSamples):
(WebCore::AudioSampleDataSource::pullSamplesInternal):
(WebCore::AudioSampleDataSource::pullAvailableSampleChunk):
(WebCore::AudioSampleDataSource::pullAvailableSamplesAsChunks):
* Source/WebCore/platform/audio/cocoa/CARingBuffer.cpp:
(WebCore::CARingBuffer::computeCapacityBytes):
(WebCore::CARingBuffer::computeSizeForBuffers):
(WebCore::CARingBuffer::initializeAfterAllocation):
(WebCore::CARingBuffer::allocate):
(WebCore::CARingBuffer::deallocate):
(WebCore::InProcessCARingBuffer::InProcessCARingBuffer):
(WebCore::InProcessCARingBuffer::flush):
(WebCore::InProcessCARingBuffer::allocateBuffers):
(WebCore::InProcessCARingBuffer::setCurrentFrameBounds):
(WebCore::InProcessCARingBuffer::getCurrentFrameBoundsWithoutUpdate):
(WebCore::InProcessCARingBuffer::currentStartFrame const):
(WebCore::InProcessCARingBuffer::currentEndFrame const):
(WebCore::CARingBufferStorageVector::CARingBufferStorageVector): Deleted.
(WebCore::CARingBuffer::CARingBuffer): Deleted.
(WebCore::CARingBuffer::~CARingBuffer): Deleted.
(WebCore::computeCapacityBytes): Deleted.
(WebCore::computeSizeForBuffers): Deleted.
(WebCore::CARingBuffer::adoptStorage): Deleted.
(WebCore::CARingBuffer::flush): Deleted.
(WebCore::CARingBufferStorageVector::allocate): Deleted.
(WebCore::CARingBufferStorageVector::flush): Deleted.
(WebCore::CARingBuffer::setCurrentFrameBounds): Deleted.
(WebCore::CARingBufferStorageVector::setCurrentFrameBounds): Deleted.
(WebCore::CARingBuffer::getCurrentFrameBoundsWithoutUpdate): Deleted.
(WebCore::CARingBufferStorageVector::getCurrentFrameBounds): Deleted.
(WebCore::CARingBuffer::currentStartFrame const): Deleted.
(WebCore::CARingBufferStorageVector::currentStartFrame const): Deleted.
(WebCore::CARingBuffer::currentEndFrame const): Deleted.
(WebCore::CARingBufferStorageVector::currentEndFrame const): Deleted.
(WebCore::CARingBuffer::updateFrameBounds): Deleted.
* Source/WebCore/platform/audio/cocoa/CARingBuffer.h:
(WebCore::CARingBuffer::updateFrameBounds):
(WebCore::CARingBuffer::frameOffset const):
(WebCore::CARingBufferStorage::updateFrameBounds): Deleted.
(WebCore::CARingBuffer::storage): Deleted.
* Source/WebCore/platform/graphics/avfoundation/AudioSourceProviderAVFObjC.mm:
(WebCore::AudioSourceProviderAVFObjC::AudioSourceProviderAVFObjC):
* Source/WebKit/GPUProcess/media/RemoteAudioDestinationManager.cpp:
(WebKit::RemoteAudioDestinationManager::audioSamplesStorageChanged):
* Source/WebKit/GPUProcess/media/RemoteAudioDestinationManager.h:
* Source/WebKit/GPUProcess/media/RemoteAudioDestinationManager.messages.in:
* Source/WebKit/GPUProcess/media/RemoteAudioSourceProviderProxy.cpp:
(WebKit::RemoteAudioSourceProviderProxy::createRingBuffer):
* Source/WebKit/GPUProcess/media/RemoteAudioSourceProviderProxy.h:
* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.h:
* Source/WebKit/GPUProcess/webrtc/RemoteAudioMediaStreamTrackRendererInternalUnitManager.cpp:
(WebKit::RemoteAudioMediaStreamTrackRendererInternalUnitManager::startUnit):
(WebKit::RemoteAudioMediaStreamTrackRendererInternalUnitManager::Unit::start):
* Source/WebKit/GPUProcess/webrtc/RemoteAudioMediaStreamTrackRendererInternalUnitManager.h:
* Source/WebKit/GPUProcess/webrtc/RemoteAudioMediaStreamTrackRendererInternalUnitManager.messages.in:
* Source/WebKit/GPUProcess/webrtc/RemoteMediaRecorder.cpp:
(WebKit::RemoteMediaRecorder::RemoteMediaRecorder):
(WebKit::RemoteMediaRecorder::audioSamplesStorageChanged):
* Source/WebKit/GPUProcess/webrtc/RemoteMediaRecorder.h:
* Source/WebKit/GPUProcess/webrtc/RemoteMediaRecorder.messages.in:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/Shared/Cocoa/SharedCARingBuffer.cpp: Renamed from Source/WebKit/Shared/Cocoa/SharedRingBufferStorage.cpp.
(WebKit::SharedCARingBufferBase::data):
(WebKit::SharedCARingBufferBase::sharedFrameBounds const):
(WebKit::SharedCARingBufferBase::getCurrentFrameBoundsWithoutUpdate):
(WebKit::SharedCARingBufferBase::flush):
(WebKit::SharedCARingBufferBase::updateFrameBounds):
(WebKit::SharedCARingBufferBase::size const):
(WebKit::ConsumerSharedCARingBuffer::ConsumerSharedCARingBuffer):
(WebKit::ConsumerSharedCARingBuffer::map):
(WebKit::ConsumerSharedCARingBuffer::allocateBuffers):
(WebKit::ProducerSharedCARingBuffer::setStorage):
(WebKit::ProducerSharedCARingBuffer::allocateBuffers):
(WebKit::ProducerSharedCARingBuffer::deallocateBuffers):
(WebKit::ProducerSharedCARingBuffer::setCurrentFrameBounds):
* Source/WebKit/Shared/Cocoa/SharedCARingBuffer.h: Renamed from Source/WebKit/Shared/Cocoa/SharedRingBufferStorage.h.
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.cpp:
(WebKit::UserMediaCaptureManagerProxy::SourceProxy::~SourceProxy):
* Source/WebKit/UIProcess/SpeechRecognitionRemoteRealtimeMediaSource.cpp:
(WebKit::m_manager):
(WebKit::SpeechRecognitionRemoteRealtimeMediaSource::setStorage):
(WebKit::m_ringBuffer): Deleted.
* Source/WebKit/UIProcess/SpeechRecognitionRemoteRealtimeMediaSource.h:
* Source/WebKit/UIProcess/SpeechRecognitionRemoteRealtimeMediaSourceManager.cpp:
(WebKit::SpeechRecognitionRemoteRealtimeMediaSourceManager::setStorage):
* Source/WebKit/UIProcess/SpeechRecognitionRemoteRealtimeMediaSourceManager.h:
* Source/WebKit/UIProcess/SpeechRecognitionRemoteRealtimeMediaSourceManager.messages.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/media/RemoteAudioDestinationProxy.cpp:
(WebKit::RemoteAudioDestinationProxy::RemoteAudioDestinationProxy):
* Source/WebKit/WebProcess/GPU/media/RemoteAudioDestinationProxy.h:
* Source/WebKit/WebProcess/GPU/media/RemoteAudioSourceProviderManager.cpp:
(WebKit::RemoteAudioSourceProviderManager::audioStorageChanged):
(WebKit::RemoteAudioSourceProviderManager::RemoteAudio::RemoteAudio):
(WebKit::RemoteAudioSourceProviderManager::RemoteAudio::setStorage):
* Source/WebKit/WebProcess/GPU/media/RemoteAudioSourceProviderManager.h:
* Source/WebKit/WebProcess/GPU/media/RemoteAudioSourceProviderManager.messages.in:
* Source/WebKit/WebProcess/GPU/webrtc/AudioMediaStreamTrackRendererInternalUnitManager.cpp:
(WebKit::AudioMediaStreamTrackRendererInternalUnitManager::Proxy::start):
* Source/WebKit/WebProcess/GPU/webrtc/AudioMediaStreamTrackRendererInternalUnitManager.messages.in:
* Source/WebKit/WebProcess/GPU/webrtc/MediaRecorderPrivate.cpp:
(WebKit::MediaRecorderPrivate::startRecording):
* Source/WebKit/WebProcess/GPU/webrtc/MediaRecorderPrivate.h:
* Source/WebKit/WebProcess/Speech/SpeechRecognitionRealtimeMediaSourceManager.cpp:
(WebKit::SpeechRecognitionRealtimeMediaSourceManager::Source::Source):
(WebKit::SpeechRecognitionRealtimeMediaSourceManager::Source::~Source):
(WebKit::SpeechRecognitionRealtimeMediaSourceManager::Source::storage): Deleted.
* Source/WebKit/WebProcess/cocoa/RemoteCaptureSampleManager.cpp:
(WebKit::RemoteCaptureSampleManager::audioStorageChanged):
(WebKit::RemoteCaptureSampleManager::RemoteAudio::setStorage):
* Source/WebKit/WebProcess/cocoa/RemoteCaptureSampleManager.h:
* Source/WebKit/WebProcess/cocoa/RemoteCaptureSampleManager.messages.in:
* Source/WebKit/WebProcess/cocoa/RemoteRealtimeMediaSourceProxy.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/CARingBuffer.cpp:
(TestWebKitAPI::CARingBufferTest::SetUp):

Canonical link: <a href="https://commits.webkit.org/256273@main">https://commits.webkit.org/256273@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/33e52bc99e5a5e5f6a5b88b6cafebfff1313f449

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/95276 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/4558 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/28251 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/104869 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/165135 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/99265 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/4543 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/33272 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/87640 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/100755 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/100941 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/3303 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/81815 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/30322 "Passed tests") | 
| [❌ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/98865 "webkitpy-tests (failure)") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/85204 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/87063 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/73222 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/39000 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/18620 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/36808 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/19906 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4334 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/40759 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/42572 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/42743 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/39151 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

eric-carlson commented on Source/WebKit/Shared/Cocoa/SharedCARingBuffer.cpp:
```diff
@@ -24,83 +24,98 @@
+auto SharedCARingBufferBase::sharedFrameBounds() const -> FrameBounds*
 {
-    return m_storage ? reinterpret_cast<FrameBounds*>(m_storage->data()) : nullptr;
+    return reinterpret_cast<FrameBounds*>(m_storage->data());
```
Other methods (e.g. `SharedCARingBufferBase::data()`, `SharedCARingBufferBase::size()`, etc) still null check `m_storage`, is there really no need to do so here?

kkinnunen-apple replied:
Great eyes. Yes, went ahead of myself and then backtracked and then made the mistake.
Added the null check.

eric-carlson reviewed (approved):
r=me

kkinnunen-apple committed:
Constructing shared CARingBuffer cannot fail
https://bugs.webkit.org/show_bug.cgi?id=247360
rdar://problem/101856691

Reviewed by Eric Carlson.

This is a problem because shared CARingBuffers are used cross-process,
and the WP might send data that would construct a potentially invalid
CARingBuffer. This leads to code that uses empty instances instead of
not checking all use-sites correctly. This in turn leads to running
code that is expected to only run when the ring buffer is correct.

Instead, make ConsumerCARingBuffer construction failable and hold
the instance accordingly. This is similar to many other related,
regular WebKit concepts like SharedMemory. Check all the construction
sites and use sites to not start audio processing unless the ring buffer is
present. This makes it easier in future to add more IPC validation
to the call sites that pass the ring buffer around and ensure that
the other parts of the code are consistent.

Previously, CARingBuffer was non-polymorphic class and its storage
class CARingBufferStorage was the polymorphic base class.
The usage was:
 CARingBuffer with CARingBufferStorageVector for in process ring buffer
 CARingBuffer with ReadOnlyRingBufferStorage for GPUP side consumer
 CARingBuffer with SharedRingBufferStorage for WP side producer

The two-class split does not make sense, as there are only one
non-polymorphic class and the arity is 1:1 in all cases.

Instead, rename:
 CARingBuffer is polymorphic base class
 InProcessCARingBuffer is the in-process ring buffer
 ProducerSharedCARingBuffer is the WP producer
 ConsumerSharedCARingBuffer is the read only GPUP side consumer

Moves the WebKit only code, CARingBuffer::adoptStorage() to
ConsumerSharedCARingBuffer::map().

Make sure Consumer is not superclass of Producer, as they're not
substitutable.

Use type alias ConsumerSharedCARingBuffer::Handle to communicate
the ring buffer to the receiver. This is so that it is obvious
what is being transferred as well as future-proofing the case
in which transferring ConsumerSharedCARingBuffer would need
more than the SHM handle. Use rvalue references to pass
the handle.

* Source/WebCore/platform/audio/cocoa/AudioSampleDataSource.h:
* Source/WebCore/platform/audio/cocoa/AudioSampleDataSource.mm:
(WebCore::AudioSampleDataSource::AudioSampleDataSource):
(WebCore::AudioSampleDataSource::setOutputFormat):
(WebCore::AudioSampleDataSource::pushSamplesInternal):
(WebCore::AudioSampleDataSource::pullSamples):
(WebCore::AudioSampleDataSource::pullSamplesInternal):
(WebCore::AudioSampleDataSource::pullAvailableSampleChunk):
(WebCore::AudioSampleDataSource::pullAvailableSamplesAsChunks):
* Source/WebCore/platform/audio/cocoa/CARingBuffer.cpp:
(WebCore::CARingBuffer::computeCapacityBytes):
(WebCore::CARingBuffer::computeSizeForBuffers):
(WebCore::CARingBuffer::initializeAfterAllocation):
(WebCore::CARingBuffer::allocate):
(WebCore::CARingBuffer::deallocate):
(WebCore::InProcessCARingBuffer::InProcessCARingBuffer):
(WebCore::InProcessCARingBuffer::flush):
(WebCore::InProcessCARingBuffer::allocateBuffers):
(WebCore::InProcessCARingBuffer::setCurrentFrameBounds):
(WebCore::InProcessCARingBuffer::getCurrentFrameBoundsWithoutUpdate):
(WebCore::InProcessCARingBuffer::currentStartFrame const):
(WebCore::InProcessCARingBuffer::currentEndFrame const):
(WebCore::CARingBufferStorageVector::CARingBufferStorageVector): Deleted.
(WebCore::CARingBuffer::CARingBuffer): Deleted.
(WebCore::CARingBuffer::~CARingBuffer): Deleted.
(WebCore::computeCapacityBytes): Deleted.
(WebCore::computeSizeForBuffers): Deleted.
(WebCore::CARingBuffer::adoptStorage): Deleted.
(WebCore::CARingBuffer::flush): Deleted.
(WebCore::CARingBufferStorageVector::allocate): Deleted.
(WebCore::CARingBufferStorageVector::flush): Deleted.
(WebCore::CARingBuffer::setCurrentFrameBounds): Deleted.
(WebCore::CARingBufferStorageVector::setCurrentFrameBounds): Deleted.
(WebCore::CARingBuffer::getCurrentFrameBoundsWithoutUpdate): Deleted.
(WebCore::CARingBufferStorageVector::getCurrentFrameBounds): Deleted.
(WebCore::CARingBuffer::currentStartFrame const): Deleted.
(WebCore::CARingBufferStorageVector::currentStartFrame const): Deleted.
(WebCore::CARingBuffer::currentEndFrame const): Deleted.
(WebCore::CARingBufferStorageVector::currentEndFrame const): Deleted.
(WebCore::CARingBuffer::updateFrameBounds): Deleted.
* Source/WebCore/platform/audio/cocoa/CARingBuffer.h:
(WebCore::CARingBuffer::updateFrameBounds):
(WebCore::CARingBuffer::frameOffset const):
(WebCore::CARingBufferStorage::updateFrameBounds): Deleted.
(WebCore::CARingBuffer::storage): Deleted.
* Source/WebCore/platform/graphics/avfoundation/AudioSourceProviderAVFObjC.mm:
(WebCore::AudioSourceProviderAVFObjC::AudioSourceProviderAVFObjC):
* Source/WebKit/GPUProcess/media/RemoteAudioDestinationManager.cpp:
(WebKit::RemoteAudioDestinationManager::audioSamplesStorageChanged):
* Source/WebKit/GPUProcess/media/RemoteAudioDestinationManager.h:
* Source/WebKit/GPUProcess/media/RemoteAudioDestinationManager.messages.in:
* Source/WebKit/GPUProcess/media/RemoteAudioSourceProviderProxy.cpp:
(WebKit::RemoteAudioSourceProviderProxy::createRingBuffer):
* Source/WebKit/GPUProcess/media/RemoteAudioSourceProviderProxy.h:
* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.h:
* Source/WebKit/GPUProcess/webrtc/RemoteAudioMediaStreamTrackRendererInternalUnitManager.cpp:
(WebKit::RemoteAudioMediaStreamTrackRendererInternalUnitManager::startUnit):
(WebKit::RemoteAudioMediaStreamTrackRendererInternalUnitManager::Unit::start):
* Source/WebKit/GPUProcess/webrtc/RemoteAudioMediaStreamTrackRendererInternalUnitManager.h:
* Source/WebKit/GPUProcess/webrtc/RemoteAudioMediaStreamTrackRendererInternalUnitManager.messages.in:
* Source/WebKit/GPUProcess/webrtc/RemoteMediaRecorder.cpp:
(WebKit::RemoteMediaRecorder::RemoteMediaRecorder):
(WebKit::RemoteMediaRecorder::audioSamplesStorageChanged):
* Source/WebKit/GPUProcess/webrtc/RemoteMediaRecorder.h:
* Source/WebKit/GPUProcess/webrtc/RemoteMediaRecorder.messages.in:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/Shared/Cocoa/SharedCARingBuffer.cpp: Renamed from Source/WebKit/Shared/Cocoa/SharedRingBufferStorage.cpp.
(WebKit::SharedCARingBufferBase::data):
(WebKit::SharedCARingBufferBase::sharedFrameBounds const):
(WebKit::SharedCARingBufferBase::getCurrentFrameBoundsWithoutUpdate):
(WebKit::SharedCARingBufferBase::flush):
(WebKit::SharedCARingBufferBase::updateFrameBounds):
(WebKit::SharedCARingBufferBase::size const):
(WebKit::ConsumerSharedCARingBuffer::ConsumerSharedCARingBuffer):
(WebKit::ConsumerSharedCARingBuffer::map):
(WebKit::ConsumerSharedCARingBuffer::allocateBuffers):
(WebKit::ProducerSharedCARingBuffer::setStorage):
(WebKit::ProducerSharedCARingBuffer::allocateBuffers):
(WebKit::ProducerSharedCARingBuffer::deallocateBuffers):
(WebKit::ProducerSharedCARingBuffer::setCurrentFrameBounds):
* Source/WebKit/Shared/Cocoa/SharedCARingBuffer.h: Renamed from Source/WebKit/Shared/Cocoa/SharedRingBufferStorage.h.
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.cpp:
(WebKit::UserMediaCaptureManagerProxy::SourceProxy::~SourceProxy):
* Source/WebKit/UIProcess/SpeechRecognitionRemoteRealtimeMediaSource.cpp:
(WebKit::m_manager):
(WebKit::SpeechRecognitionRemoteRealtimeMediaSource::setStorage):
(WebKit::m_ringBuffer): Deleted.
* Source/WebKit/UIProcess/SpeechRecognitionRemoteRealtimeMediaSource.h:
* Source/WebKit/UIProcess/SpeechRecognitionRemoteRealtimeMediaSourceManager.cpp:
(WebKit::SpeechRecognitionRemoteRealtimeMediaSourceManager::setStorage):
* Source/WebKit/UIProcess/SpeechRecognitionRemoteRealtimeMediaSourceManager.h:
* Source/WebKit/UIProcess/SpeechRecognitionRemoteRealtimeMediaSourceManager.messages.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/media/RemoteAudioDestinationProxy.cpp:
(WebKit::RemoteAudioDestinationProxy::RemoteAudioDestinationProxy):
* Source/WebKit/WebProcess/GPU/media/RemoteAudioDestinationProxy.h:
* Source/WebKit/WebProcess/GPU/media/RemoteAudioSourceProviderManager.cpp:
(WebKit::RemoteAudioSourceProviderManager::audioStorageChanged):
(WebKit::RemoteAudioSourceProviderManager::RemoteAudio::RemoteAudio):
(WebKit::RemoteAudioSourceProviderManager::RemoteAudio::setStorage):
* Source/WebKit/WebProcess/GPU/media/RemoteAudioSourceProviderManager.h:
* Source/WebKit/WebProcess/GPU/media/RemoteAudioSourceProviderManager.messages.in:
* Source/WebKit/WebProcess/GPU/webrtc/AudioMediaStreamTrackRendererInternalUnitManager.cpp:
(WebKit::AudioMediaStreamTrackRendererInternalUnitManager::Proxy::start):
* Source/WebKit/WebProcess/GPU/webrtc/AudioMediaStreamTrackRendererInternalUnitManager.messages.in:
* Source/WebKit/WebProcess/GPU/webrtc/MediaRecorderPrivate.cpp:
(WebKit::MediaRecorderPrivate::startRecording):
* Source/WebKit/WebProcess/GPU/webrtc/MediaRecorderPrivate.h:
* Source/WebKit/WebProcess/Speech/SpeechRecognitionRealtimeMediaSourceManager.cpp:
(WebKit::SpeechRecognitionRealtimeMediaSourceManager::Source::Source):
(WebKit::SpeechRecognitionRealtimeMediaSourceManager::Source::~Source):
(WebKit::SpeechRecognitionRealtimeMediaSourceManager::Source::storage): Deleted.
* Source/WebKit/WebProcess/cocoa/RemoteCaptureSampleManager.cpp:
(WebKit::RemoteCaptureSampleManager::audioStorageChanged):
(WebKit::RemoteCaptureSampleManager::RemoteAudio::setStorage):
* Source/WebKit/WebProcess/cocoa/RemoteCaptureSampleManager.h:
* Source/WebKit/WebProcess/cocoa/RemoteCaptureSampleManager.messages.in:
* Source/WebKit/WebProcess/cocoa/RemoteRealtimeMediaSourceProxy.cpp:
* Tools/TestWebKitAPI/Tests/WebCore/CARingBuffer.cpp:
(TestWebKitAPI::CARingBufferTest::SetUp):

Canonical link: https://commits.webkit.org/256273@main
